app/controllers/network_services_controller.go

```diff
@@ -375,6 +379,159 @@ func ensureMasqueradeIptablesRule(masqueradeAll bool, podCidr string) error {
 	return nil
 }
 
+// syncHairpinIptablesRules adds/removes iptables rules pertaining to traffic
+// from an Endpoint (Pod) to its own service VIP. Rules are only applied if
+// enabled globally via CLI argument or a service has an annotation requesting
+// it.
+func (nsc *NetworkServicesController) syncHairpinIptablesRules() error {
```

Comment:
I am assuming we will call this function only if gloabal flag is set or atleast one service is annotated. Right?